cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_delete.spec.js

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+const miscUtils = new MiscUtils(cy);
+const workspace1Name = 'test_workspace_320sdfouAz';
+let workspace1Description = 'This is a workspace1 description.';
+const workspace2Name = 'test_workspace_321sdfouAz';
+let workspace2Description = 'This is a workspace2 description.';
+
+let workspace1Id;
+let workspace2Id;
+
+if (Cypress.env('WORKSPACE_ENABLED')) {
+  describe('Delete Workspace(s) in 2 ways in workspace list page', () => {
+    before(() => {
+      Cypress.config('defaultCommandTimeout', 60000);
+      cy.deleteAllWorkspaces();
+    });
+    beforeEach(() => {
+      // Visit workspace list page
+      miscUtils.visitPage(`/app/workspace_list`);
+      cy.createWorkspace({
+        name: workspace1Name,
+        description: workspace1Description,
+        features: ['workspace_detail', 'use-case-observability'],
+        settings: {
+          permissions: {
+            library_write: { users: ['%me%'] },
+            write: { users: ['%me%'] },
+          },
+        },
+      }).then((value) => {
+        workspace1Id = value;
+        cy.intercept('DELETE', `/api/workspaces/${workspace1Id}`).as(
+          'deleteWorkspace1Request'
+        );
+      });
+
+      cy.createWorkspace({
+        name: workspace2Name,
+        description: workspace2Description,
+        features: ['workspace_detail', 'use-case-search'],
+        settings: {
+          permissions: {
+            library_write: { users: ['%me%'] },
+            write: { users: ['%me%'] },
+          },
+        },
+      }).then((value) => {
+        workspace2Id = value;
+        cy.intercept('DELETE', `/api/workspaces/${workspace2Id}`).as(
+          'deleteWorkspace2Request'
+        );
+      });
+    });
+
+    afterEach(() => {
+      cy.deleteAllWorkspaces();
+    });
+
+    describe('delete a workspace successfully using action buttons', () => {
+      it('should successfully load delete button and show delete modal when clicking action button', () => {
+        cy.contains(workspace1Name).should('be.visible');
+        cy.getElementByTestId('euiCollapsedItemActionsButton').first().click();
```

Comment:
Is there any other way to find this button?

Reply:
can we use ```cy.getElementByTestId(`checkboxSelectRow-${workspaceId}`)```

Reply:
Thank you for all the comments, updated.